### ISSUE TITLE
White noise in speakers when idle

### ISSUE THREAD

**Issue by TranzistorP** [linuxmint/linuxmint#677]
Pulseaudio disable speaker output for power saving when they are not in use, causing the cable to act as an antenna and pick up white noise. I think this is a common issue that many people encounter.

https://forums.linuxmint.com/viewtopic.php?t=320488
https://askubuntu.com/questions/1275299/strange-hissing-sound-from-speakers-when-idle
https://forum.ixbt.com/topic.cgi?id=12:57870
https://www.youtube.com/watch?v=YP_FQLciZRI

Disabling the module-suspend-on-idle module solves this problem.

`pactl unload-module module-suspend-on-idle`

This command disables this module until the system is rebooted.
I added it to startup.

But this is not user-friendly.

There are two possible solutions to this issue:

1) Add the ability to disable this module in the GUI. 
2) Disable this module by default, except when the device is running on battery power

**Comment by fredcw** [comment]
Agreed. I think this is a big issue for Linux Mint as it doesn't make a good first impression, and for newbies who don't know how to fix it, they may well just try a different distro because of it. I don't know how much power is saves but I don't imagine it's a lot, and in any case, I don't think anyone wants an audible hum even if it does save power.

I don't think it needs to be an option, just the default, including laptops.

**Comment by billyswong** [comment]
The solution I have been using for a long time is similar to what descibed in https://forums.linuxmint.com/viewtopic.php?t=307575

My computer currently contains a file `/etc/modprobe.d/hda.conf`, with content
```
options snd-hda-intel power_save=0 power_save_controller=N
```

I guess this solution/workaround can be better than disable the suspend-on-idle globally, as audio output through hdmi or dp cable won't pop or pick up white noise as they are digital.

PulseAudio is being replaced by Pipewire. `/etc/pulse/default.pa` may not work in the future. The distant future of `pactl` is also unknown.

**Comment by fluorescent-haze** [comment]
I agree. Just had to explain to someone on the Linux Mint support channel how to use nano to edit `/etc/modprobe.d/` -- not fun. This should probably be part of the `kmod` package like many other `modprobe.d` files.

**Comment by TranzistorP** [comment]
After switching to PipeWare, the old solution broke. 
But it became possible to disable power saving for each device individually!

New solution:

1) Create directory /home/user/.config/wireplumber/main.lua.d
` mkdir -p ~/.config/wireplumber/main.lua.d`

2) Place the file 51-some_script_name.lua with the following contents in it:

```
rule = {
  matches = {
    {
      { "node.description", "equals", "Built-in Audio Analog Stereo" },
    },
  },
  apply_properties = {
    ["session.suspend-timeout-seconds"] = 0,
  },
}
table.insert(alsa_monitor.rules, rule)
```
This disables power saving only for my regular audio output, Built-in Audio Analog Stereo.
 If you need to disable power saving for another device, then replace Built-in Audio Analog Stereo with the name of your device.
 
 The names can be viewed with the command:

`wpctl status`

<details>
  <summary>It will output something like this:</summary>

```
PipeWire 'pipewire-0' [1.0.5, tranzi@tranzi-MS-7D22, cookie:3747526747]
 
 └─ Clients:
        32. pipewire                            [1.0.5, tranzi@tranzi-MS-7D22, pid:1147]
        47. cinnamon                            [1.0.5, tranzi@tranzi-MS-7D22, pid:22080]
        52. WirePlumber                         [1.0.5, tranzi@tranzi-MS-7D22, pid:22062]
        55. cinnamon-settings.py                [1.0.5, tranzi@tranzi-MS-7D22, pid:22080]
        64. Cinnamon Volume Control Media Keys  [1.0.5, tranzi@tranzi-MS-7D22, pid:1398]
        65. Blueman                             [1.0.5, tranzi@tranzi-MS-7D22, pid:1660]
        66. Cinnamon Volume Control             [1.0.5, tranzi@tranzi-MS-7D22, pid:1573]
        67. Muffin                              [1.0.5, tranzi@tranzi-MS-7D22, pid:1573]
        68. xdg-desktop-portal                  [1.0.5, tranzi@tranzi-MS-7D22, pid:1996]
        72. Firefox                             [1.0.5, tranzi@tranzi-MS-7D22, pid:1929]
       104. wpctl                               [1.0.5, tranzi@tranzi-MS-7D22, pid:23201]
       112. cinnamon                            [1.0.5, tranzi@tranzi-MS-7D22, pid:23057]
       129. WirePlumber [export]                [1.0.5, tranzi@tranzi-MS-7D22, pid:22062]

Audio
 ├─ Devices:
 │      62. Built-in Audio                      [alsa]
 │      76. HDA NVidia                          [alsa]
 │      78. USB Audio Device                    [alsa]
 │  
 ├─ Sinks:
 │  *   43. Built-in Audio Analog Stereo        [vol: 0.21]
 │      98. USB Audio Device Analog Stereo      [vol: 0.50]
 │     111. HDA NVidia Digital Stereo (HDMI)    [vol: 0.40]
 │  
 ├─ Sink endpoints:
 │  
 ├─ Sources:
 │  *   97. USB Audio Device Mono               [vol: 1.00]
 │     107. Built-in Audio Analog Stereo        [vol: 1.00]
 │  
 ├─ Source endpoints:
 │  
 └─ Streams:
        33. cinnamon                                                    
             38. monitor_MONO   
             79. input_MONO      < USB Audio Device:capture_MONO	[active]
       114. cinnamon                                                    
            101. monitor_MONO   
            113. input_MONO      < USB Audio Device:capture_MONO	[active]

Video
 ├─ Devices:
 │  
 ├─ Sinks:
 │  
 ├─ Sink endpoints:
 │  
 ├─ Sources:
 │  
 ├─ Source endpoints:
 │  
 └─ Streams:

Settings
 └─ Default Configured Node Names:
         0. Audio/Sink    alsa_output.pci-0000_00_1f.3.analog-stereo  
```

</details>

After this, restart wireplumber with the command:

`systemctl --user restart wireplumber`


But at the moment there is a bug - it is necessary to play some sound with this device after a restart and only then the noise will stop until the next system reboot.

And I am still sure that this functionality needs to be included in the GUI Sound Settings.

**Comment by fredcw** [comment]
What worked for me (I'm using pipewire on manjaro) is creating the file:

`/etc/modprobe.d/audio_disable_powersave.conf`

with contents:
```
options snd_hda_intel power_save=0
```
It's the top answer on https://askubuntu.com/questions/1230833/annoying-click-popping-sound-on-ubuntu-20-04
